apps/rwa/app/api/profile/route.ts

```diff
@@ -1,15 +1,53 @@
+import { withApiAuthRequired, getAccessToken } from "@auth0/nextjs-auth0";
 import type { Profile } from "profile-service";
 import { NextResponse } from "next/server";
+import { cookies } from "next/headers";
 import { getProfile as getProfile_ } from "profile-service";
 
-export const GET = async (): Promise<NextResponse<Profile>> => {
-  const data = await getProfile_();
+interface ErrorResponse {
+  code: string;
+  status: number;
+}
```

Comment:
Lots of opportunity for abstractions in this file. Should look into a well defined error handling pattern.